### PR TITLE
Sparse/cufe

### PR DESCRIPTION
**Pull request by ZiaddAhmedd**
team name: cufe

### CODE REVIEW

**Review by ingberam — CHANGES_REQUESTED** [review]
Hello, the code structure looks OK. But please remove the results files from the PR.

The code builds and runs on a VM with 8cpu/16GB memory (GCP n2-custom-8-16384).

@ZiaddAhmedd please approve that these results are roughly what you would expect.

```
cufe,cufe_linscan,sparse-full,10,228.63062740751894,0.0,807.9983868598938,6202796.0,0,0,sparse,0.4767191977077364
cufe,cufe_linscan,sparse-full,10,186.17849232775876,0.0,807.9983868598938,6202796.0,0,0,sparse,0.7357879656160458
cufe,cufe_linscan,sparse-full,10,128.15355242620848,0.0,807.9983868598938,6202796.0,0,0,sparse,0.8635386819484241
cufe,cufe_linscan,sparse-full,10,97.44300309584673,0.0,807.9983868598938,6202796.0,0,0,sparse,0.8900716332378223
cufe,cufe_linscan,sparse-full,10,94.59717776195376,0.0,807.9983868598938,6202796.0,0,0,sparse,0.8953581661891118
cufe,cufe_linscan,sparse-full,10,104.95061285654172,0.0,807.9983868598938,6202796.0,0,0,sparse,0.9040687679083094
cufe,cufe_linscan,sparse-full,10,92.67380212115589,0.0,807.9983868598938,6202796.0,0,0,sparse,0.9045558739255014
cufe,cufe_linscan,sparse-full,10,90.086083840717,0.0,807.9983868598938,6202796.0,0,0,sparse,0.9086532951289398
cufe,cufe_linscan,sparse-full,10,68.74486880655289,0.0,807.9983868598938,6202796.0,0,0,sparse,0.9518194842406876
cufe,cufe_linscan,sparse-full,10,40.838583958549016,0.0,807.9983868598938,6202796.0,0,0,sparse,0.9959025787965616
```

**Comment by ZiaddAhmedd** [comment]
I removed the results file from the PR.
Yes, the results are close to the expected.

File: results-out.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+writing output to results/sparse-full.png
```
Comment (ingberam) [inline results-out.txt:1]: please remove this file

Comment (AbdelrahmanMohamed129) [reply results-out.txt:1]: It is already deleted on my repo this line is written on plotting right?

Comment (ingberam) [reply results-out.txt:1]: I meant delete `results-out.txt`

Comment (AbdelrahmanMohamed129) [reply results-out.txt:1]: oh okay sorry for the inconvenience

**Review by ingberam — APPROVED** [review]

**Pull request merged**